lib/services/http-service.js

```diff
@@ -145,26 +146,43 @@ function httpServiceFactory(
     HttpService.prototype.createSwagger = function() {
         var app = this.app;
         var endpoint = this.endpoint;
+
+        function loadYamlConfig(yamlPath) {
+            var config = yaml.safeLoad(fs.readFileSync(yamlPath, 'utf8'));
```

Comment:
Change to use promise version of readFile, as `createSwagger` is using promise, and so that the read file error can be passed out of `createSwagger` via promise rejection.